src/clj_kondo/impl/analyzer.clj

```diff
@@ -2115,30 +2115,59 @@
 
-(defn lint-keyword-call! [ctx kw namespaced? arg-count expr]
+(defn analyze-keyword-call [ctx kw namespaced? arg-count expr]
   (let [callstack (:callstack ctx)
-        config (:config ctx)]
-    (when-not (config/skip? config :invalid-arity callstack)
-      (let [ns (:ns ctx)
-            ?resolved-ns (if namespaced?
-                           (if-let [kw-ns (namespace kw)]
-                             (or (get (:qualify-ns ns) (symbol kw-ns))
-                                 ;; because we couldn't resolve the namespaced
-                                 ;; keyword, we print it as is
-                                 (str ":" (namespace kw)))
-                             ;; if the keyword is namespace, but there is no
-                             ;; namespace, it's the current ns
-                             (:name ns))
-                           (namespace kw))
-            kw-str (if ?resolved-ns (str ?resolved-ns "/" (name kw))
-                       (str (name kw)))]
-        (when (or (zero? arg-count)
-                  (> arg-count 2))
-          (findings/reg-finding! ctx
-                                 (node->line (:filename ctx) expr :invalid-arity
-                                             (format "keyword :%s is called with %s args but expects 1 or 2"
-                                                     kw-str
-                                                     arg-count))))))))
+        config (:config ctx)
+        ns (:ns ctx)
+        ?resolved-ns (if namespaced?
+                       (if-let [kw-ns (namespace kw)]
+                         (or (get (:qualify-ns ns) (symbol kw-ns))
+                             ;; because we couldn't resolve the namespaced
+                             ;; keyword, we print it as is
+                             (str ":" (namespace kw)))
+                         ;; if the keyword is namespace, but there is no
+                         ;; namespace, it's the current ns
+                         (:name ns))
+                       (namespace kw))
+        kw-str (if ?resolved-ns (str ?resolved-ns "/" (name kw))
+                   (str (name kw)))]
+    (when (and (not (config/skip? config :invalid-arity callstack))
+               (or (zero? arg-count)
+                   (> arg-count 2)))
+      (findings/reg-finding! ctx
+                             (node->line (:filename ctx) expr :invalid-arity
+                                         (format "keyword :%s is called with %s args but expects 1 or 2"
+                                                 kw-str
+                                                 arg-count))))
+    ;; With one argument to the keyword call, we can check if the return is a map.
+    ;; Another linter warning we could create is to check if the 2-arity version
+    ;; will never return the default value.
+    (when (= arg-count 1)
+      (let [analyzed (first (analyze-children (update ctx :callstack conj [nil :list]) (rest (:children expr))))
```

Comment:
Here I want to check the return type of the unique argument, is there a better way to do it?

Reply:
This works most of the time but isn't always reliable. More reliable is to insert an :id on the node you're interested in and then check the `:id` on the returned value. This is done elsewhere in the code.

Reply:
Great! Will take a look on it later, thanks o/